mono/metadata/gc.c

```diff
@@ -221,7 +221,7 @@ mono_gc_run_finalize (void *obj, void *data)
 	runtime_invoke (o, NULL, &exc, NULL);
 
 	if (exc) {
-		/* fixme: do something useful */
+		mono_unhandled_exception(exc);
```

Comment:
Looks like it was never implemented any other way - the /*fixme*/ came in as part of adding support for mono_runtime_invoke to catch exceptions: https://github.com/Unity-Technologies/mono/commit/7cb732d788da5d938849e8f923efeb1672460761